### ISSUE TITLE
Updating .sats names

### ISSUE THREAD

**Issue by larrysalibra** [satsnames/satsnamesapi#4]
Names serve as human-readable, memorable pointers to information or objects. Over time, users will want to point their names at more information. This is how names in DNS work. You can register a name and point it a different resources - servers, services, etc - over its lifetime.

`.sats` names would be much more useful and valuable to their owners and people building services in need of human-readable, user-custodied, secure identifiers. I'm starting this thread as a continuation of a [twitter thread](https://twitter.com/larrysalibra/status/1636377543799615488) to brainstorm ideas about how we can update our names.

One idea I proposed is that users update their names by signing a message that proves ownership of the ordinal that owns the name. They could then make a new inscription using any address with a json object that includes this signature, the name being updated and new keys to add to the name:

```json
{ 
"p": "sns", 
"op": "up", 
"auth": "insert bip-322 sig here",
"name": "satoshi.sats", 
"key-to-be-added-or-updated": "new value",
"key-to-be-removed": null
}
```

This would require wallet support for BIP-322 signed messages and the ability to sign these for addresses that own ordinals.

What other ideas do people have?

**Comment by andrewtoth** [comment]
Interesting. It's unclear to me where the message being signed is though. Should the message be the entire op? I think that would be safest. However, in that case the witness should be segregated ;).
Also, should we consider changing the protocol to binary instead of json? Using a binary protocol would save a lot in blockchain space and therefore fees. It would also prevent json parsing quirks from different libraries causing issues.

**Comment by RagnarLifthrasir** [comment]
Thanks for starting this @larrysalibra  👍🏻

Signing a message to prove ownership of an Inscription has advantages. But it doesn't solve for duplicate names. I could use ten different addresses to inscribe larry.sats ten times. Which is the valid one? "First is first" is a social contract, not a smart contract or otherwise enforced by (software) protocol. Like BNS for example.

Also, how does one build a reputation or link to activities and apps? I can sign in to various Stacks apps with my Hiro or Xverse wallet that has my BNS name. And Trajan...

**Comment by RagnarLifthrasir** [comment]
And slightly off-topic but related is this:
https://github.com/hirosystems/wallet/issues/3459

**Comment by RagnarLifthrasir** [comment]
With Bitcoin or blockchain-based naming systems, remember Zooko's triangle; it's incredibly challenging for a single kind of name to achieve more than two of the three necessary properties.
https://en.wikipedia.org/wiki/Zooko%27s_triangle

![1920px-Zooko's_Triangle svg](https://user-images.githubusercontent.com/24993355/226948398-6ca4c00c-c5b3-42a0-8f27-7eef256ec064.png)

**Comment by bensig** [comment]
It would be very useful to resolve .sats names to display various types of data: 

- data in another inscription: image, html, markdown, text
- forward to a web address
- forward to a .onion address
- show basic info about the .sats name including the owner, mint date, and any marketplace listings

I have started putting some ideas down and I'm working on a chrome plugin and a subdomain to show some of these concepts. Should have something to demo next week.

**Comment by scribblesats** [comment]
@bensig Feel free to reach out if you would like to share any of what you're working on, or if you're just looking for someone to test drive ideas. I'm a mod in the Sats Names discord. Not a dev myself but happy to offer a "community" perspective to you and @sats-wizard. Cheers.

**Comment by OliverThompsonOT** [comment]
> The current Ordinals .sats name lacks the _security_ property because anyone can duplicate a name.

you are wrong, now you can not mint a Sats Names called "001.sats"

if you can, you are right, but you cannot. eventhough you mint a nft has this names, but no one acknowledge your name.

it's safe.

**Comment by RagnarLifthrasir** [comment]
> > The current Ordinals .sats name lacks the _security_ property because anyone can duplicate a name.
> 
> you are wrong, now you can not mint a Sats Names called "001.sats"
> 
> if you can, you are right, but you cannot. eventhough you mint a nft has this names, but no one acknowledge your name.
> 
> it's safe.

You are wrong. How can you not know that ANYONE can inscribe any .sats name? Some explorers and inscription services display the number of duplicate names! And anyone can create an explorer or application that acknowledges any .sats name according to any rule.

**Comment by OliverThompsonOT** [comment]
> > > The current Ordinals .sats name lacks the _security_ property because anyone can duplicate a name.
> > 
> > 
> > you are wrong, now you can not mint a Sats Names called "001.sats"
> > if you can, you are right, but you cannot. eventhough you mint a nft has this names, but no one acknowledge your name.
> > it's safe.
> 
> You are wrong. How can you not know that ANYONE can inscribe any .sats name? Some explorers and inscription services display the number of duplicate names! And anyone can create an explorer or application that acknowledges any .sats name according to any rule.

just like bitcoin, someone said that BSV is the real bitcoin, so what?
it is self-deception.

also, you can make a bitcoin wallelt that it shows your address have 1000 bitcoin, and you sell them to others, you can do it, but so what? no one believe you.

maybe some marketplace is not good, but we can verify them , just like bitcoin, **don't trust, just verify.**

if you buy ens, it's real? but who tell you it's real?

**Comment by OliverThompsonOT** [comment]
> > > The current Ordinals .sats name lacks the _security_ property because anyone can duplicate a name.
> > 
> > 
> > you are wrong, now you can not mint a Sats Names called "001.sats"
> > if you can, you are right, but you cannot. eventhough you mint a nft has this names, but no one acknowledge your name.
> > it's safe.
> 
> You are wrong. How can you not know that ANYONE can inscribe any .sats name? Some explorers and inscription services display the number of duplicate names! And anyone can create an explorer or application that acknowledges any .sats name according to any rule.

"anyone can create an explorer or application that acknowledges any .sats name according to any rule."

it just like that some one can create an explorer or application that acknowledges BSV/BCH is real bitcoin. 
good lucky.
